translation-service-eTranslation/src/main/java/eu/europeana/api/translation/service/etranslation/ETranslationTranslationService.java

```diff
@@ -0,0 +1,224 @@
+import eu.europeana.api.translation.definitions.model.TranslationObj;
+import eu.europeana.api.translation.service.AbstractTranslationService;
+import eu.europeana.api.translation.service.exception.TranslationException;
+import eu.europeana.api.translation.service.util.UtilityMethods;
+
+public class ETranslationTranslationService extends AbstractTranslationService {
+  
+  private static final Logger LOGGER = LogManager.getLogger(ETranslationTranslationService.class);
+
+  private String serviceId;
+  private final String baseUrl;
+  private final String domain;
+  //this is the base url of the translation api (without the request handler (or the controller) endpoint)
+  private final String callbackUrl;
+  private final String credentialUsername;
+  private final String credentialPwd;
+  private final int maxWaitMillisec;
+  private final RedisMessageListenerContainer redisMessageListenerContainer;
+  public static final String baseUrlTests="base-url-for-testing";
+  //base64 encoded string, which will not be translated from eTransaltion, used as a markup delimiter
+//  public static final String markupDelimit=" ZXd3ZHdld2U= ";
+  public static final String markupDelimit="\n###==###\n";
+  
+  public ETranslationTranslationService(String baseUrl, String domain, String callbackUrl, int maxWaitMillisec, 
+      String username, String password, RedisMessageListenerContainer redisMessageListenerContainer) throws TranslationException {
+    if(!baseUrlTests.equals(baseUrl) && (StringUtils.isBlank(baseUrl) || StringUtils.isBlank(domain) || StringUtils.isBlank(callbackUrl) ||
+        maxWaitMillisec<=0 || StringUtils.isBlank(username) || StringUtils.isBlank(password))) {
+      throw new TranslationException("Invalid eTranslation config parameters.");
+    }
+    this.baseUrl = baseUrl;
+    this.domain = domain;
+    this.callbackUrl=callbackUrl;
+    this.maxWaitMillisec=maxWaitMillisec;
+    this.credentialUsername=username;
+    this.credentialPwd=password;
+    this.redisMessageListenerContainer=redisMessageListenerContainer;
+  }
+
+  @Override
+  public void translate(List<TranslationObj> translationObjs) throws TranslationException {
+    String eTranslJointStr = generateJointStringForTranslation(translationObjs, markupDelimit);
+    
+    /* create an eTransl request with an external reference and send it. The same external reference is received
+     * in the eTransl callback. That reference is used for the name of the channel for the redis message subscriber 
+     * listener created below, which will be notified from the redis publisher after the eTransl callback comes.
+     * The publisher will publish to the same channel using the external reference from the eTransl callback.
+     */
+    //create external reference for eTransl service
+    String eTranslExtRef = UtilityMethods.generateRedisKey(eTranslJointStr, translationObjs.get(0).getSourceLang(), translationObjs.get(0).getTargetLang());
```

Comment:
we have to add a prefix to diferentiate between the regular keys for translation and the ones for etranslation requests. please appenf the "et:" prefix. The prefix can be added as paramenter to the generateRedisKey method